packages/scandipwa/src/component/Menu/Menu.component.tsx

```diff
@@ -149,17 +149,28 @@ export class MenuComponent extends PureComponent<MenuComponentProps> {
     }
 
     renderSubLevel(category: FormattedMenuItem, isSecondLevel = false): ReactElement {
-        const { activeMenuItemsStack, device } = this.props;
+        const {
+            activeMenuItemsStack, device, visitedMenuItems, updateVisitedMenuItems,
+        } = this.props;
         const { item_id, children, title } = category;
         const childrenArray = getSortedItems(Object.values(children));
         const isVisible = activeMenuItemsStack.includes(item_id);
         const subcategoryMods = { type: 'subcategory' };
+        const isColapsed = !isVisible && visitedMenuItems.includes(item_id);
+
+        if (isVisible && !visitedMenuItems.includes(item_id)) {
+            updateVisitedMenuItems([...visitedMenuItems, item_id]);
+        }
+
+        if (isColapsed && visitedMenuItems.includes(item_id)) {
+            updateVisitedMenuItems(visitedMenuItems.filter((id) => id !== item_id));
+        }
```

Comment:
no business logic in render methods, please refactor the implementation by moving this to the container.